src/Sql/Sql/ManagedInstanceDtc/Services/AzureSqlManagedInstanceDtcAdapter.cs

```diff
@@ -0,0 +1,79 @@
+using Microsoft.Azure.Management.Sql.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedInstanceDtc.Services
+{
+    public class AzureSqlManagedInstanceDtcAdapter
+    {
+        /// <summary>
+        /// Gets or sets the AzureEndpointsCommunicator which has all the needed management clients
+        /// </summary>
+        private AzureSqlManagedInstanceDtcCommunicator Communicator { get; set; }
+
+        /// <summary>
+        /// Gets or sets the Azure profile
+        /// </summary>
+        public IAzureContext Context { get; set; }
+
+        public AzureSqlManagedInstanceDtcAdapter(IAzureContext context)
+        {
+            Context = context;
+            Communicator = new AzureSqlManagedInstanceDtcCommunicator(Context);
+        }
+
+        public AzureSqlManagedInstanceDtcModel GetManagedInstanceDtc(string resourceGroup, string managedInstanceName)
+        {
+            var resp = Communicator.Get(resourceGroup, managedInstanceName);
+            return CreateManagedInstanceDtcModelFromResponse(resourceGroup, managedInstanceName, resp);
+        }
+
+        public AzureSqlManagedInstanceDtcModel SetManagedInstanceDtc(string resourceGroup, string managedInstanceName, AzureSqlManagedInstanceDtcModel dtcModel )
+        {
+            Management.Sql.Models.ManagedInstanceDtc parameters = new Management.Sql.Models.ManagedInstanceDtc()
+            {
+                DtcEnabled = dtcModel.DtcEnabled,
+                ExternalDnsSuffixSearchList = dtcModel.ExternalDnsSuffixSearchList,
+                SecuritySettings = dtcModel.SecuritySettings,
+
+            };
+
+            var resp = Communicator.CreateOrUpdate(resourceGroup, managedInstanceName, parameters);
+            return CreateManagedInstanceDtcModelFromResponse(resourceGroup, managedInstanceName, resp);
+        }
+
+        private AzureSqlManagedInstanceDtcModel CreateManagedInstanceDtcModelFromResponse(string resourceGroup, string managedInstanceName, Management.Sql.Models.ManagedInstanceDtc managedInstanceDtc)
+        {
+            AzureSqlManagedInstanceDtcModel managedInstanceDtcModel = new AzureSqlManagedInstanceDtcModel();
+            managedInstanceDtcModel.ManagedInstanceName = managedInstanceName;
+            managedInstanceDtcModel.Id = managedInstanceDtc.Id;
+            managedInstanceDtcModel.ResourceGroupName= resourceGroup;
+            managedInstanceDtcModel.DtcEnabled = managedInstanceDtc.DtcEnabled.GetValueOrDefault();
+
+            // Due to a bug in DTC workflows, the full host name is returned as a DTC host name DNS suffix.
+            // However, we can extract the suffix from the host name.
+            managedInstanceDtcModel.DtcHostName = managedInstanceDtc.DtcHostNameDnsSuffix;
+            managedInstanceDtcModel.DtcHostNameDnsSuffix = managedInstanceDtc.DtcHostNameDnsSuffix.Substring(managedInstanceDtc.DtcHostNameDnsSuffix.IndexOf('.') + 1);
```

Comment:
what happens when they fix the bug, can we ensure this 'hack' doesn't get forgotten and we pass back incomplete dns suffix?
e.g. could we come up with an assert that's guaranteed to fail if we bump API version and start receiving correct dns suffix from backend

Reply:
I've added a check for the DtcHostName property, which determines the logic for setting the property in the response